### PR TITLE
Fix AirTerminal:singleDuct:ConstantVolume:FourPipeBeam Minimum OA Reference

### LINKED ISSUE #6479: AirTerminal:singleDuct:ConstantVolume:FourPipeBeam not responding correctly to minimum ventilation limit when autosize design primary airflow
Issue overview
--------------
AirTerminal:singleDuct:ConstantVolume:FourPipeBeam not responding correctly to minimum ventilation limit when user selects autosize design primary airflow and autosize chilled water flow, hot water flow, and total beam length. 

When attempting to demonstrate lack of effectiveness of fourpipe beams in a ventilation driven lab, it was discovered that the primary airflow for the fourpipebeam does not correctly account for minimum ventilation if ventilation is larger than airflow required for heating/cooling the space. 
Examining the example fourpipebeam file, the condition can be replicated, whereas a DesignSpecification:OutdoorAir for zone 'Perimeter_mid_ZN_1 4pipe Beam' is changed to AirChangesperHour and set to be a large value > cooling or heating airflow. The primary airflow remains unchanged whether 10 ACH, 100 ACH, 1000 ACH, etc. is input as a ventilation requirement (hourly peak at 14640 cfm). The  supply airflow rate reported in ComponentSizing report remains unchanged at 6.91 m3/s, while the air flow rate adjustment increases corresponding to required ventilation input. 
image
image
image
image

Currently working around the issue by calculating the primary airflow required for ventilation, setting the design primary airflow for the beam accordingly. This requires an iterative process when a mix of ventilation driven and load driven spaces are examined with fourpipebeams in a model. Autosizing primary airflow is not working. A working autosizing condition will hold minimum of ventilation flow requested in the zone. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system: WINDOWS 10 Enterprise)
 - Version of EnergyPlus (v8.8)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by matthew-larson**
Pull request overview
---------------------
 - Fixes #6479 
 - This pull request fixes the issue where the zone OA requirement was using .DesOAFlow, which is looking at only the Outdoor Air Flow per Zone specified in the DesignSpecification:OutdoorAir object and ignoring the other Outdoor Air Flow inputs (per Person, per Zone Floor Area, and Changes per Hour).  The resolution now uses .MinOA, which is the design minimum outside air in m3/s from the zone sizing calculation.
 
### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally

### CODE REVIEW

**Comment by mjwitte** [comment]
@matthew-larson Whacky timing. I just (minutes ago) stumbled onto this while working on Space. 
I'm deleting `.DesOAFlow` from the sizing structs and found it lurking in fourpipe. So this is very timely.
Now to make sure I'm not breaking `TermUnitFinalZoneSizing.MinOA`.

File: src/EnergyPlus/HVACFourPipeBeam.cc [hunk 1]
```diff
@@ -855,7 +855,7 @@ namespace FourPipeBeam {
             Real64 minFlow(0.0);
             Real64 maxFlowCool(0.0);
             minFlow = std::min(state.dataEnvrn->StdRhoAir * originalTermUnitSizeMaxVDot,
-                               state.dataSize->TermUnitFinalZoneSizing(state.dataSize->CurTermUnitSizingNum).DesOAFlow * state.dataEnvrn->StdRhoAir);
+                               state.dataSize->TermUnitFinalZoneSizing(state.dataSize->CurTermUnitSizingNum).MinOA * state.dataEnvrn->StdRhoAir);
```
Comment (rraustad) [inline src/EnergyPlus/HVACFourPipeBeam.cc]: I don't even see where DesOAFlow is set in current develop. So was this always 0 prior to this change?

Comment (rraustad) [reply src/EnergyPlus/HVACFourPipeBeam.cc]: Now I am confused, I don' even see where DesOAFlow is a member of ZoneSizingData which is where TermUnitFinalZoneSizing comes from.

    EPVector<DataSizing::ZoneSizingData> TermUnitFinalZoneSizing; // Final data for sizing terminal units (indexed per terminal unit)

Comment (matthew-larson) [reply src/EnergyPlus/HVACFourPipeBeam.cc]: @mjwitte was deleting this from the sizing structs when working on `Space` (mentioned in comment above) so this was getting removed anyways.  Looks like he has done this at this point.  The value was 0 before unless a Flow/Zone was used for the outdoor air specification, which was what DesOAFlow referenced previously.

**Comment by Myoldmopar** [comment]
Sooo.....this is just a new unit test change now?  I have not tried to digest the conversation above except that it looked like you were about to fix something and @mjwitte was removing it.  If this is just the unit test change at this point and nothing is left then I am happy to merge it right now, CI was super green as of yesterday.

**Comment by matthew-larson** [comment]
That's basically what's happening, we both came across the issue at different angles and his merge ended up implementing my same fix.

**Comment by Myoldmopar** [comment]
Great, log this as completed and I'll test the unit test with develop pulled in and merge this.  Thanks!

**Comment by Myoldmopar** [comment]
Builds and passes locally.  Merging.  Thanks @matthew-larson and @mjwitte for the combined effort :+1:

**Pull request merged**